app/models/gamification_leaderboard.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class DiscourseGamification::GamificationLeaderboard < ::ActiveRecord::Base
+  self.table_name = 'gamification_leaderboards'
+
+  def self.scores_for(leaderboard_name) 
+    leaderboard = self.find_by(name: leaderboard_name)
+    leaderboard.to_date ||= Date.today
```

Comment:
```suggestion
    leaderboard.to_date ||= DiscourseGamification::GamificationScore.minimum(:date)
```

Reply:
I don't think this is what we want...
```
=> #<DiscourseGamification::GamificationLeaderboard:0x00000001286b3130
 id: 3,
 name: "test",
 from_date: Tue, 22 Mar 2022,
 to_date: nil,
```

I think a structure like this could be expected.... a user wants a leaderboard from the 22nd (from_date) to Today (so we leave to_date nil)

from_date => Tue, 22 Mar 2022,

DiscourseGamification::GamificationScore.minimum(:date) => Fri, 18 Mar 2022
vs 
Date.today => Fri, 25 Mar 2022

Reply:
Oh that was my bad, I miss read it as the `FROM` date instead of `TO`.

Reply:
I meant for us to handle a NIL FROM as picking stuff from the beginning. How are you handling a missing FROM at the moment?

Reply:
```
users = users.where("gamification_scores.date BETWEEN ? AND ?", leaderboard.from_date, leaderboard.to_date) if leaderboard.from_date.present?
```

If from date is not present we just grab all records